MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -3629,8 +3629,15 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
 		id<MTLDevice> sysMTLDevice = MTLCreateSystemDefaultDevice();
 		if (mvkGetRegistryID(sysMTLDevice) == mvkGetRegistryID(mtlDevice)) {
 			_physicalDevice->replaceMTLDevice(sysMTLDevice);
+			mtlDevice = _physicalDevice->getMTLDevice();
```

Comment:
I don't have the hardware to test the above scenario but this is required if replaceMTLDevice changes the _mtlDevice variable

Reply:
This is performed as part of creating a swap chain, which really isn't the appropriate place to do set the shader compilation options. The notes to the commit for the replacement code seems to indicate the pointer won't actually change anyway.

So, let's move the code setting `shouldMaximizeConcurrentCompilation` to the `MVKPhysicalDevice` constructor. This is the first time we're actually configuring something on the `MTLDevice`, so I suggest we create a member function called something like `initMTLDevice()`, and call it in the `MVKPhysicalDevice` constructor, above the call to `initProperties()`.

------------------- 

@cdavis5e In reviewing your note from your older commit, it indicates the `MTLDevice` pointers are the same. If that's the case, is the call to `replaceMTLDevice:` even needed? Based on your note, if I understand the flow correctly, the `MTLDevice` pointer will always be the same as what's already in `MVKPhysicalDevice`? If that's not the case, then a lot could have happened with the old `MTLDevice` (settings and possible resources) that will now be out of scope?

Reply:
They should be the same. Frankly, I don't see how Metal could work on Mac if they *weren't* the same. From what I understand, Metal creates `MTLDevice` objects for all GPUs when it initializes; the array of those devices is what `MTLCopyAllDevices()` returns. All `MTLCreateSystemDefaultDevice()` does on Mac is:
1. force Macs with a mux to select the high-performance GPU for presentation instead of the integrated one, and
2. return a +1 reference to the "default" device, which on Macs with a mux will be the high-performance GPU, and on Macs with multiple video cards will be the device for the primary display.

Reply:
Thanks for your feedback, I've created https://github.com/KhronosGroup/MoltenVK/pull/1945 based on your suggestions but I'm unable to test this. Sorry.

Reply:
> Thanks for your feedback, I've created #1945 based on your suggestions but I'm unable to test this. Sorry.

Thanks. The call to `MTLCreateSystemDefaultDevice()` in #1945 needs work. Since this PR is otherwise the same as #1945, I'm going to pull this one in.